pallets/oracle/src/lib.rs

```diff
@@ -251,26 +220,29 @@ pub mod pallet {
 impl<T: Config> Pallet<T> {
 	// public only for testing purposes
 	pub fn begin_block(_height: T::BlockNumber) {
-		// read to a temporary value, because we can't alter the map while we iterate over it
-		let raw_values_updated: Vec<_> = RawValuesUpdated::<T>::iter().collect();
+		let oracle_keys: Vec<_> = OracleKeys::<T>::get();
 
 		let current_time = Self::get_current_time();
 
 		let mut updated_items = Vec::new();
-		for (key, is_updated) in raw_values_updated.iter() {
-			if *is_updated || Self::is_outdated(key, current_time) {
-				let new_value = Self::update_aggregate(key);
-				updated_items.push((key.clone(), new_value));
+		let max_delay = Self::get_max_delay();
+		for key in oracle_keys.iter() {
+			let price = Self::get_timestamped(key);
+			let Some(price) = price else{
+				continue;
+			};
+			let is_outdated = current_time > price.timestamp + max_delay;
+			if !is_outdated {
+				updated_items.push((key.clone(), Some(price.value)));
 			}
 		}
-
+		let updated_items_len = updated_items.len();
 		if !updated_items.is_empty() {
 			Self::deposit_event(Event::<T>::AggregateUpdated { values: updated_items });
 		}
 
 		let current_status_is_online = Self::is_oracle_online();
-		let new_status_is_online = !raw_values_updated.is_empty() &&
-			raw_values_updated.iter().all(|(key, _)| Aggregate::<T>::get(key).is_some());
+		let new_status_is_online = oracle_keys.len() > 0 && updated_items_len == oracle_keys.len();
```

Comment:
Adding the change in my previous comment implies that we also need to change this logic. If we allow the 'push' of `None` values to the updated_items vec, we need to check here if all items actually have a value of `Some(x)` before we set the status back to online.

Reply:
removed option 
```
AggregateUpdated {
			values: Vec<(OracleKey, T::UnsignedFixedPoint)>,
		},
```
there is no need for option in AggregateUpdated pallet event